src/runtime-rs/crates/hypervisor/src/device/device_manager.rs

```diff
@@ -0,0 +1,338 @@
+// Copyright (c) 2019-2022 Alibaba Cloud
+// Copyright (c) 2019-2022 Ant Group
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use crate::DeviceType::{Block, Undefined, Vfio};
+use crate::{
+    device::blk_dev_manager::BlockDeviceManager,
+    device_type::{Device, DeviceArgument, GenericConfig},
+    DeviceManagerInner, DeviceType, Hypervisor,
+};
+use anyhow::{anyhow, Context, Result};
+use ini::Ini;
+use std::{collections::HashMap, sync::Arc};
+use tokio::sync::{Mutex, RwLock};
+
+pub type ArcBoxDevice = Arc<Mutex<Box<dyn Device>>>;
+const SYS_DEV_PREFIX: &str = "/sys/dev";
+
+/// VIRTIO_BLOCK_MMIO indicates block driver is virtio-mmio based
+pub(crate) const VIRTIO_BLOCK_MMIO: &str = "virtio-blk-mmio";
+/// VIRTIO_BLOCK_PCI indicates block driver is virtio-pci based
+pub(crate) const VIRTIO_BLOCK_PCI: &str = "virtio-blk-pci";
+pub(crate) const KATA_MMIO_BLK_DEV_TYPE: &str = "mmioblk";
+pub(crate) const KATA_BLK_DEV_TYPE: &str = "blk";
+
+/// block_index and released_block_index are used to search an available block index
+/// in Sandbox.
+///
+/// @block_index generally default is 1 for <vdb>;
+/// @released_block_index for blk devices removed and indexes will released at the same time.
+#[derive(Clone, Debug, Default)]
+struct SharedInfo {
+    block_index: u64,
+    released_block_index: Vec<u64>,
+}
+
+impl SharedInfo {
```

Comment:
The `SharedInfo` looks like only for the block device manager, which I think should not move to the block device manager.

Reply:
The `sharedInfo` could also work for VFIO manger